datadog-logs-oci-orm/modules/containerregistry/login.tf

```diff
@@ -0,0 +1,91 @@
+resource "null_resource" "get_username" {
+    provisioner "local-exec" {
+        command = <<EOT
+            #!/bin/bash
+            set -e
+
+            # Obtain the username from the current_user_ocid
+            echo "Obtaining username from current_user_ocid..."
+            username=$(oci iam user get --user-id ${var.current_user_ocid} --query "data.name" --raw-output)
+
+            echo $username > /tmp/username.txt
+        EOT
+    }
+    triggers = {
+        always_run = "${timestamp()}"
+    }
+}
+
+resource "null_resource" "recreate_auth_token" {
+    provisioner "local-exec" {
+        command = <<EOT
+            #!/bin/bash
+            set -e
+            if [ "${var.auth_token}" != "" ]; then
+                echo "Using existing auth token..."
+                echo ${var.auth_token} > /tmp/new_auth_token.txt
+            else
+                # Step 1: List existing auth tokens
+                echo "Listing existing auth tokens..."
+                existing_token_ocid=$(oci iam auth-token list --user-id ${var.current_user_ocid} \
+                    --query "data[?description=='${var.auth_token_description}'].id | [0]" --raw-output)
+
+                if [ "$existing_token_ocid" != "" ]; then
+                    echo "Deleting existing auth token: $existing_token_ocid"
+                    oci iam auth-token delete --user-id ${var.current_user_ocid} --auth-token-id $existing_token_ocid --force
+                else
+                    # Check the total number of existing tokens
+                    total_tokens=$(oci iam auth-token list --user-id ${var.current_user_ocid} --query "length(data)" --raw-output)
+                    
+                    if [ "$total_tokens" -eq 2 ]; then
+                        echo "Error: Total existing tokens are equal to 2. Cannot create a new token."
+                        exit 1
+                    fi
+                    echo "No existing auth token with description '${var.auth_token_description}' found. Proceeding to create a new one."
+                fi
+
+                # Step 2: Create a new auth token
+                echo "Creating a new auth token..."
+                new_token_value=$(oci iam auth-token create --user-id ${var.current_user_ocid} \
+                    --description "${var.auth_token_description}" --query "data.token" --raw-output)
+
+                # Step 3: Sleep for 30 seconds to ensure token propagation
+                echo "Waiting for 30 seconds to ensure token availability..."
+                sleep 30
+
+                echo $new_token_value > /tmp/new_auth_token.txt
```

Comment:
nit ⛏️  I'm curious on what infra does this text file gets saved? Reason I bring it up is because if for some reason, the next resource call doesn't run and it doesn't get deleted, it could be a security risk